core/src/com/unciv/ui/tilegroups/TileGroup.kt

```diff
@@ -321,18 +321,17 @@ open class TileGroup(
                 i += 1
             }
             val finalLocation = existingImages.random(Random(tileInfo.position.hashCode() + locationToCheck.hashCode()))
-
             val image = ImageGetter.getImage(finalLocation)
-            tileBaseImages.add(image)
-            baseLayerGroup.addActorAt(0,image)
-            setHexagonImageSize(image)
-        }
 
-        if (tileBaseImages.isEmpty()) { // Absolutely nothing! This is for the 'default' tileset
-            val image = ImageGetter.getImage(tileSetStrings.hexagon)
             tileBaseImages.add(image)
             baseLayerGroup.addActor(image)
-            setHexagonImageSize(image)
+
+            if (tileSetStrings.tileSetConfig.tileScales.isNotEmpty()) {
+                val scale = tileSetStrings.tileSetConfig.tileScales[tileBaseImageLocations[index].takeLastWhile { it != '/' }]
+                setHexagonImageSize(image, scale)
+            } else {
+                setHexagonImageSize(image)
+            }
```

Comment:
Technically you can write this as
```kotlin
val scale = tileSetStrings.tileSetConfig.tileScales[tileBaseImageLocations[index].takeLastWhile { it != '/' }]
setHexagonImageSize(image, scale)
```
So you don't need the if statement, but I'm not sure it improves readability enough to merit this change.

Reply:
I wanted to avoid that takeLastWhile is called for every Image (location) even though it's unnecessary